Client/core/CMainMenu.cpp

```diff
@@ -221,6 +224,47 @@ CMainMenu::CMainMenu(CGUI* pManager)
     m_pLatestNews->SetProperty("InheritsAlpha", "False");
     m_pLatestNews->SetVisible(false);
 
+    const std::pair<const char*, GUI_CALLBACK> socialButtonDatas[]{
+        {"website", GUI_CALLBACK(&CMainMenu::OnSocialMediaButtonClick, this)},
+        {"discord", GUI_CALLBACK(&CMainMenu::OnSocialMediaButtonClick, this)},
+        {"twitter", GUI_CALLBACK(&CMainMenu::OnSocialMediaButtonClick, this)},
+        {"facebook", GUI_CALLBACK(&CMainMenu::OnSocialMediaButtonClick, this)},
+        {"youtube", GUI_CALLBACK(&CMainMenu::OnSocialMediaButtonClick, this)},
+        {"github", GUI_CALLBACK(&CMainMenu::OnSocialMediaButtonClick, this)},
+    };
+
+    m_socialMediaLinks = {
+        {"website", "https://multitheftauto.com"},
+        {"discord", "https://discord.gg/mtasa"},
+        {"twitter", "https://twitter.com/mtaqa"},
+        {"facebook", "https://www.facebook.com/multitheftauto"},
+        {"youtube", "https://www.youtube.com/user/mtaqa"},
+        {"github", "https://github.com/multitheftauto/mtasa-blue"},
+    };
+
+    float offsetX = 0.01f;
+    size_t i = 0;
+    const CVector2D buttonSize((NATIVE_SOCIAL_ICON_X / NATIVE_RES_X) * m_iMenuSizeX, (NATIVE_SOCIAL_ICON_Y / NATIVE_RES_Y) * m_iMenuSizeY);
+    for (const auto& [name, buttonClickHandler] : socialButtonDatas)
+    {
+        CGUIStaticImage* button(pManager->CreateStaticImage(m_pCanvas));
+
+        button->LoadFromFile(SString("cgui\\images\\socialset\\%s.png", name).c_str());
+        button->SetProperty("InheritsAlpha", "False");
+        button->SetAlpha(0.35f);
+        button->SetSize(buttonSize, false);
+        button->SetPosition({offsetX * m_iMenuSizeX - buttonSize.fX / 2, 0.97f * m_iMenuSizeY - buttonSize.fY / 2}, false);
+        button->SetZOrderingEnabled(false);
+        button->SetMouseEnterHandler(GUI_CALLBACK(&CMainMenu::OnSocialMediaButtonHover, this));
+        button->SetMouseLeaveHandler(GUI_CALLBACK(&CMainMenu::OnSocialMediaButtonUnhover, this));
+        button->SetClickHandler(buttonClickHandler);
+
+        m_socialButtonLinks[button] = m_socialMediaLinks[name];
+
+        m_socialButtons[i++] = button;
+        offsetX += 0.03f;
+    }
```

Comment:
You should use lambda functions for the event handlers. You avoid the static methods in CMainMenu class and it keeps the handler functions close to the point of creation. Furthermore, you could've stored the URL in the CEGUI element user data field.
```cpp
    size_t socialsIndex = 0;

    for (const auto& nameAndLink : socialMediaLinks)
    {
        CGUIStaticImage* button = pManager->CreateStaticImage(m_pCanvas);
        m_socialsImages[socialsIndex++].reset(button);
        button->LoadFromFile(SString("cgui\\images\\socialset\\%s.png", nameAndLink.first.c_str()).c_str());
        button->SetProperty("InheritsAlpha", "False");
        button->SetAlpha(0.35f);
        // button->SetSize();
        // button->SetPosition();
        button->SetZOrderingEnabled(false);
        button->SetUserData(const_cast<char*>(nameAndLink.second.c_str()));
        button->SetMouseEnterHandler(GUI_CALLBACK(
            [](CGUIElement* button)
            {
                button->SetAlpha(0.08f);
                return true;
            }));
        button->SetMouseLeaveHandler(GUI_CALLBACK(
            [](CGUIElement* button)
            {
                button->SetAlpha(0.35f);
                return true;
            }));
        button->SetClickHandler(GUI_CALLBACK(
            [](CGUIElement* button)
            {
                auto url = static_cast<const char*>(button->GetUserData());
                ShellExecuteNonBlocking("open", url);
                return true;
            }));
    }
```